src/Rendering_Internals.md

```diff
@@ -0,0 +1,410 @@
+1. [Push pixels](<#Push pixels>)
+
+[^fetch_size]:
+Well, since pixels are 2bpp, two fetches are necessary—but one still ends up with 8 pixels, since the VRAM data bus is 8-bit.
+
+### Get tile ID
+
+This step determines which background/window tile to fetch pixels from.
+This step is not executed by the Pixel Slice Fetcher itself, but rather by the active "fetcher":
+
+#### BG fetcher
+
+During this step, a tilemap is sampled to determine which tile to fetch.
+
+The address read depends on whether the BG fetcher is in ["BG mode" or "Window mode"](<#>):
+<table style="text-align: center;" class="compact">
+   <thead>
+      <tr>
+         <th>Mode</th><th>15</th><th>14</th><th>13</th><th>12</th><th>11</th><th>10</th><th>9</th><th>8</th><th>7</th><th>6</th><th>5</th><th>4</th><th>3</th><th>2</th><th>1</th><th>0</th>
+      </tr>
+   </thead>
+   <tbody>
+      <tr>
+         <th>BG</th><td rowspan="2">1</td><td rowspan="2">0</td><td rowspan="2">0</td><td rowspan="2">1</td><td rowspan="2">1</td><td>LCDC bit 3</td><td colspan="5">(LY + SCY) ∕ 8</td><td colspan="5">(LX + SCX) ∕ 8</td>
+      </tr>
+      <tr>
+         <th>Window</th><td>LCDC bit 6</td><td colspan="5">"Window Y" ∕ 8</td><td colspan="5">LX ∕ 8</td>
+      </tr>
+   </tbody>
+</table>
+
+::: tip Wrapping
+
+The two additions in the "BG mode" row are carried out in 8 bits, i.e. modulo 256.
+Due the division by 8, the modulo is essentially 32, i.e. a tilemap's width.
+This is what causes the background to wrap around, both vertically and horizontally.
+
+:::
+
+A byte is read from the computed address, and is forwarded to the Pixel Slice Fetcher as a tile ID.
+Color models read from the computed address twice: from VRAM bank 0 to get the tile ID, and from VRAM bank 1 to get the attributes.[^banks_parallel]
+
+::: tip Raster effects
+
+Interestingly, unlike e.g. the NES' PPU, great care has been taken to ensure that the BG fetcher re-reads as many registers as possible (`SCY`, `LCDC`, etc.).
+This may have been insight from the former console, on which [proper "raster splits" are quite tricky](https://www.nesdev.org/wiki/PPU_scrolling#Split_X_scroll) due to a lot of internal caching.
+
+:::
+
+This step takes 2 dots, with the VRAM access(es) being performed on the second.
+
+[^banks_parallel]:
+Both appear to be accessed during the same dot, which implies that the PPU can access both VRAM banks at the same time.
+
+#### OBJ fetcher
+
+Since the Pixel Slice Fetcher is normally used continuously for the BG FIFO, the OBJ fetcher waits to take control until two conditions are met:
+- The Pixel Slice Fetcher is attempting to [push pixels](<#Push pixels>)
+- The BG FIFO is not empty
+
+Once both conditions are fulfilled, the OBJ FIFO takes over, discarding the pixels slices already fetched.
+Note that if the BG FIFO is empty, the Pixel Slice Fetcher immediately switches to [Get tile ID](<#Get tile ID>) when refilling it, so the OBJ fetcher will wait for 6 additional dots.
```

Comment:
According to my understanding of the sprite timing, the maximum "penalty" for a sprite occurring too soon in the BG slice fetch sequence is 5 dots (making that sprite contribute at most 11 dots to mode 3). I'm not clear on the technical reason behind this, though.

This understanding is corroborated by https://gbdev.io/pandocs/STAT.html#properties-of-stat-modes and https://www.reddit.com/r/EmuDev/comments/59pawp/gb_mode3_sprite_timing/